### PR TITLE
Batched SVD added (gesvdjBatched and gesvdaStridedBatched)

### PR DESCRIPTION
**Pull request by nikopj**
`gesvdjBatched` and `gesvdaStridedBatched` are now accessable for performing SVD on a sequence of matrices, represented by a 3 dimensional tensor (3rd dimension is the batch dimension). `Linalg.svd` has been overloaded accordingly for 3 dim tensors. 

- [x] tests added

### CODE REVIEW

File: lib/cusolver/dense.jl [hunk 1]
```diff
@@ -494,6 +494,131 @@ for (bname, fname, elty, relty) in ((:cusolverDnSgesvdj_bufferSize, :cusolverDnS
     end
 end
 
+for (bname, fname, elty, relty) in ((:cusolverDnSgesvdjBatched_bufferSize, :cusolverDnSgesvdjBatched, :Float32, :Float32),
+                                    (:cusolverDnDgesvdjBatched_bufferSize, :cusolverDnDgesvdjBatched, :Float64, :Float64),
+                                    (:cusolverDnCgesvdjBatched_bufferSize, :cusolverDnCgesvdjBatched, :ComplexF32, :Float32),
+                                    (:cusolverDnZgesvdjBatched_bufferSize, :cusolverDnZgesvdjBatched, :ComplexF64, :Float64))
+    @eval begin
+        function gesvdj!(jobz::Char,
+                         econ::Int,
```
Comment (maleadt) [inline lib/cusolver/dense.jl]: Shouldn't this be a simple Bool?

File: lib/cusolver/linalg.jl [hunk 2]
```diff
@@ -248,16 +251,39 @@ function _svd!(A::CuMatrix{T}, full::Bool, alg::JacobiAlgorithm) where T
     U, S, V = gesvdj!('V', Int(!full), A)
     return SVD(U, S, V')
 end
+function _svd!(A::CuArray{T,3}, full::Bool, alg::JacobiAlgorithm) where T
+    # @assert size(A,1) <= 32 && size(A,2) <= 32 # see cuSolver API for gesvdjBatched
```
Comment (maleadt) [inline lib/cusolver/linalg.jl]:
```suggestion
```

**Review by maleadt — COMMENTED** [review]
Some minor nits, but LGTM, thanks!

File: lib/cusolver/dense.jl (hunk 1 above)
Comment (nikopj) [reply lib/cusolver/dense.jl]: My mistake. I was copying the structure of the unbatched gesvdj. The batched one doesn't take an econ argument and always does full sized svd (see api [here](https://docs.nvidia.com/cuda/cusolver/#cusolverdn-t-gesvdjbatched)). It is fixed now.

**Pull request merged**